### PR TITLE
Button: Add clarification on selections

### PR DESCRIPTION
**Pull request by maximedegreve**
Following some [confusion in Slack (internally)](https://github.slack.com/archives/C01L618AEP9/p1713368314318559) regarding the display of multiple selections within our buttons, we convened to discuss and concluded that documenting this would be beneficial.

### CODE REVIEW

File: content/components/button.mdx [hunk 1]
```diff
@@ -428,6 +428,39 @@ Limit primary button usage. Only use one primary button on the page, whenever po
   </Dont>
 </DoDontContainer>
 
+### Selections
+
+Buttons are commonly used to show a choice within a `select`, [SelectPanel](/components/selectpanel) or [ActionMenu](/components/action-menu). In such instances, they are frequently paired with an internal or external label.
```
Comment (mperrotti) [inline content/components/button.mdx:433]: > to show a choice within a `select`

When do we use a button to show the selections from a `select`?

File: content/components/button.mdx [hunk 2]
```diff
@@ -462,21 +495,25 @@ Read more about [descriptive buttons](/guides/accessibility/descriptive-buttons)
...
+  src="https://github.com/primer/design/assets/2313998/b138567f-3ba8-49ab-9fa1-80a4fe30b034"
+/>
 
 </Box>
 
 This message should be in an ARIA live region, using `aria-live="polite"`. The live region must be present on page load, but the message inside the live region should only be rendered while the button is in a loading state.
 
-If an error prevents process from being completed, focus should be brought to an `<h2>` (or next relevant heading) of the error banner. 
-
+If an error prevents process from being completed, focus should be brought to an `<h2>` (or next relevant heading) of the error banner.
```
Comment (mperrotti) [inline content/components/button.mdx:515]:
```suggestion
If an error prevents a process from being completed, focus should be brought to the relevant heading in the error banner.
```

**Review by mperrotti — COMMENTED** [review]
Generally looking good! Main concern is the mention of a `select` controlling a button label.

File: content/components/button.mdx (hunk 1 above)
Comment (maximedegreve) [reply content/components/button.mdx:433]: Actually I might be wrong on this one. I can remove that.

**Review by mperrotti — APPROVED** [review]

**Pull request merged**